### ISSUE TITLE
yocto environment doesn't work anymore

### ISSUE THREAD

**Issue by wucke13** [nix-community/nix-environments#31]
[this line](https://github.com/nix-community/nix-environments/blob/868ac314e3b4e04ce5e20ae88fb4daf7b211d2a5/envs/yocto/shell.nix#L54) causes the following error:

```console
ERROR: Variable BB_ENV_EXTRAWHITE has been renamed to BB_ENV_PASSTHROUGH_ADDITIONS
ERROR: Variable BB_ENV_EXTRAWHITE from the shell environment has been renamed to BB_ENV_PASSTHROUGH_ADDITIONS
ERROR: Exiting to allow enviroment variables to be corrected
```

**Comment by Mic92** [comment]
See also https://github.com/nix-community/nix-environments/pull/26
No one has finished a fix yet.

**Comment by jhol** [comment]
A new version of the fix is provided here: #39 .

To repeat my observations attached to #26 from my testing with [OE4T (OpenEmbedded for Tegra) Demo Distro](https://github.com/OE4T/tegra-demo-distro):

There is an issue with the *shadow* recipe trying to pick up *libcrypt* from glibc. However, *libcrypt* has been removed from glibc in recent releases, and a replacement library is available: *libxcrypt*. For some reason, although *libxcrypt* is supported in the project, autotools is still trying to pick up the glibc version. As a hack I replaced `DEPENDS = "virtual/crypt"` with `DEPENDS = "libxcrypt"` in `repos/poky/meta/recipes-extended/shadow/shadow.inc`. I don't know why this is necessary, because to my understanding `virtual/crypt` should be set as an alias of `libxcrypt`.

More investigation is required. I don't know exactly what's going wrong here. I guess the system version of libcrypt leaks on Ubuntu etc., but on Nix it can't be accessed?
